cfme/fixtures/candu.py

```diff
@@ -90,3 +123,367 @@ def candu_db_restore(temp_appliance_extended_db):
+    params=['default', 'custom-hourly', 'custom-daily', 'custom-weekly', 'custom-monthly'])
+def chargeback_rates(appliance, request):
+    if request.param == 'default':
+        desc = 'Default'
+        yield desc
+    # Create new chargeback rates
+    else:
+        try:
+            desc = f'{request.param}_{fauxfactory.gen_alphanumeric()}'
+            per_time = request.param.split('-')[1]
+            rate_dict = {
+                'per_time': per_time.capitalize(),
+                'variable_rate': str(3 * rate_divisors[per_time])
+            }
+            compute = appliance.collections.compute_rates.create(
+                description=desc,
+                fields={
+                    'Used CPU': rate_dict,
+                    'Used Disk I/O': rate_dict,
+                    'Used Memory': rate_dict,
+                    'Allocated CPU Count': rate_dict,
+                    'Allocated Memory': rate_dict
+                }
+            )
+            storage = appliance.collections.storage_rates.create(
+                description=desc,
+                fields={
+                    'Used Disk Storage': rate_dict,
+                    'Allocated Disk Storage': rate_dict
+                }
+            )
+        except Exception as ex:
+            pytest.fail(
+                f'Exception while creating chargeback rates: {ex}'
+            )
+
+        yield desc
+
+        for entity in [compute, storage]:
+            try:
+                entity.delete_if_exists()
+            except Exception as ex:
+                pytest.fail(
+                    f'Exception while deleting chargeback rates: {ex}'
+                )
+
+
+@pytest.fixture(scope="module")
+def chargeback_report(appliance, vm_ownership, assign_rates):
+    """Create and queue chargeback report."""
+    owner = vm_ownership
+    title = f'{assign_rates}'
+    report_fields = [
+        'CPU Used',
+        'CPU Used Cost',
+        'Disk I/O Used',
+        'Disk I/O Used Cost',
+        'Memory Used',
+        'Memory Used Cost',
+        'Network I/O Used',
+        'Network I/O Used Cost',
+        'Storage Used',
+        'Storage Used Cost',
+        'Memory Allocated over Time Period',
+        'Memory Allocated Cost',
+        'vCPUs Allocated over Time Period',
+        'vCPUs Allocated Cost',
+        'Storage Allocated',
+        'Storage Allocated Cost',
+        'Owner'
+    ]
+    data = {
+        'menu_name': title,
+        'title': title,
+        'base_report_on': 'Chargeback for Vms',
+        'report_fields': report_fields,
+        'filter': {
+            'filter_show_costs': 'Owner',
+            'filter_owner': owner,
+            'interval_end': 'Today (partial)'
+        }
+    }
+    report = appliance.collections.reports.create(is_candu=True, **data)
+
+    logger.info(f"Queuing chargeback report {title}.")
+    report.queue(wait_for_finish=True)
+
+    if not list(report.saved_reports.all()[0].data.rows):
```

Comment:
consider assigning this chained call to a name, so its abundantly clear that its the same chain.